ex/loqui/lib/loqui/server.ex

```diff
@@ -0,0 +1,136 @@
+defmodule Loqui.Server do
+
+  defmodule Http do
+    defmodule State do
+      defstruct [:handler, :path, :transport, :socket, :buffer, :handler_opts]
+
+      def new(socket, transport, opts) do
+        %__MODULE__{
+          path: Keyword.fetch!(opts, :loqui_path),
```

Comment:
If these are required, should they be arguments instead of options?

Reply:
This is internal to the HTTP module. The external facing `Server.start_link` function has `loqui_path` as an argument.

Reply:
I think this is due to how ranch starts a listener. you get `start_link(ref, socket, transport, opts)`, and I have to pack everything in the opts.